src/apps/api/views/mapping.py

```diff
@@ -1,83 +1,102 @@
-                        
-                        # Merge dictionaries
-                        output_dict |= tol_column_dict
+            for tol_field, mapped_data in standard_mapped_data.items():
+                if standard != 'tol':
+                    field = mapped_data.get(standard, dict()).get('field', str())
+                else:
+                    field = tol_field
+                
+                # Ignore if 'field' is nonexistent
+                if field:
+                    output_list.append(field)
+
+            output_list.sort()
+            return output_list
+        
+        elif standard and project:
+            output_list = list()
+            project_fields = DTOL_EXPORT_TO_STS_FIELDS.get(project, str())
+
+            for tol_field in project_fields:
+                if standard == 'tol':  
+                    field = tol_field
+                else:  
+                    field =  standard_mapped_data.get(tol_field, dict()).get(standard, dict()).get('field', str())
+
+                # Ignore if 'field' is nonexistent
+                if field:
+                    output_list.append(field)
                             
-            output.append(output_dict)
-
-            result = finish_request(template=output, error=None, num_found=len(output[0]),return_http_response=False)
+            return output_list
         else:
-            result =  finish_request(template=None, error=None, num_found=None, return_http_response=False)
+            return list()
 
-    # Removes slashes from output
-    result = json.loads(result)
-    return result
+def get_mapped_data_from_sample_data(standard, sample_data):
+    # COPO defined fields are fields that primarily lowercase or camel case
+    copo_defined_fields =  Sample().get_custom_sample_fields()
 
-def get_mapping(request):
-    return_type = request.GET.get('return_type', "json").lower()
+    if standard not in STANDARDS:
+        return None
     
-    default_value = d_utils.convertListToString(STANDARDS)
-    standard = request.GET.get('standard', default_value)
-    standard_list = d_utils.convertStringToList(standard)
+    with open(settings.STANDARDS_MAP_FILE_PATH) as f:
+        output_list = list()
+        standard_mapped_data = json.load(f)
+       
+        for sample in sample_data:
+            output_dict = dict()
+
+            for tol_field, value in sample.items():
+                if standard == 'tol':  
+                    field = tol_field
+                else:  
+                    field =  standard_mapped_data.get(tol_field, dict()).get(standard, dict()).get('field', str())
+               
+                # If field is defined by COPO, add it to 'output_dict'
+                if tol_field.lower().startswith('copo'):
+                    field = tol_field
+                elif tol_field in copo_defined_fields:
```

Comment:
Can we do the other way round?  we have the "sample type" and we can find the sample type fields from the schema by referencing to the property "specification"  and "version" . For those non sample type fields, we treat it as copo fields.  In that way, we don't need to add "Is_builtin" and "is_custom".